Library/Homebrew/formula.rb

```diff
@@ -657,6 +663,29 @@ def skip_clean?(path)
     self.class.skip_clean_paths.include? to_check
   end
 
+  # Sometimes we accidentally install files outside prefix. After we fix that,
+  # users will get nasty link conflict error. So we create a whitelist here to
+  # allow overwriteing certain files. e.g.
+  #   link_overwrite "bin/foo", "lib/bar"
+  #   link_overwrite "share/man/man1/baz-*"
+  def link_overwrite?(path)
+    # Don't overwrite files for a fresh installation.
+    return false unless any_old_version_installed?
```

Comment:
`brew uninstall; brew install` failing is bad. The point of the whitelist is we only whitelist files we are happy with deleting. Do you think users are going to miss `gio` modules with identical names to those they have just requested be installed by Homebrew? If we really want that behaviour it should be opt-in.